ospatch/updates_windows.go

```diff
@@ -28,21 +28,46 @@ import (
 
 // SystemRebootRequired checks whether a system reboot is required.
 func SystemRebootRequired() (bool, error) {
-	reg := `SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired`
-	regPath := `HKLM:\` + reg
-	logger.Debugf("Checking if reboot required by looking at %s", regPath)
-	k, err := registry.OpenKey(registry.LOCAL_MACHINE, reg, registry.QUERY_VALUE)
+	// https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-movefileexw#remarks
+	logger.Debugf("Checking for PendingFileRenameOperations")
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, `SYSTEM\CurrentControlSet\Control\Session Manager`, registry.QUERY_VALUE)
 	if err != nil {
-		if err == registry.ErrNotExist {
-			logger.Debugf("%s does not exist, indicating no reboot is required.", regPath)
-			return false, nil
+		if err != registry.ErrNotExist {
```

Comment:
minor: for readability, you may want to mirror the `if err == nil ... else if err != ErrNotExist`. It would remove the extra nesting and add consistency. You could also refactor `if err != nil && err != registry.ErrNotExist` since you duplicate it 3 times in this function.